mask/eolearn/mask/cloud_mask.py

```diff
@@ -292,3 +297,563 @@ def get_s2_pixel_cloud_detector(threshold=0.4, average_over=4, dilation_size=2,
+    with masking operations.
+    """
+
+    def __init__(self,
+                 mono_classifier=None,
+                 multi_classifier=None,
+                 data_feature='BANDS-S2-L1C',
+                 is_data_feature='IS_DATA',
+                 all_bands=True,
+                 processing_resolution=None,
+                 max_proc_frames=11,
+                 mono_features=None,
+                 multi_features=None,
+                 mask_feature='CLM_INTERSSIM',
+                 mono_threshold=0.4,
+                 multi_threshold=0.5,
+                 average_over=1,
+                 dilation_size=1
+                ):
+        """Constructor.
+
+        :param mono_classifier: Classifier used for mono-temporal cloud detection (`s2cloudless` or equivalent).
+                                Default value: None (s2cloudless is used)
+        :type mono_classifier: sklearn Estimator that works on the 10 selected reflectance bands
+                                (`B01`, `B02`, `B04`, `B05`, `B08`, `B8A`, `B09`, `B10`, `B11`, `B12`)
+        :param multi_classifier: Classifier used for multi-temporal cloud detection.
+                                 Default value: None (SSIM-based model is used)
+        :type multi_classifier: sklearn Estimator that works on the 90 multi-temporal features:
+                                 - raw reflectance value in the target frame,
+                                 - average value within a spatial window in the target frame,
+                                 - maximum, mean and standard deviation of the structural similarity (SSIM)
+                                 - indices between a spatial window in the target frame and every other,
+                                 - minimum and mean reflectance of all available time frames,
+                                 - maximum and mean difference in reflectances between the target frame and every other.
+        :param data_feature: Name of the key in the `eopatch.data` dictionary, which stores raw reflectance data.
+                             Default value:  `'BANDS-S2-L1C'`.
+        :type data_feature: str
+        :param is_data_feature: Name of the key in the `eopatch.mask` dictionary, which indicates whether data is valid.
+                                Default value: `'IS_DATA'`.
+        :type is_data_feature: str
+        :param all_bands: Flag, which indicates whether images will consist of all 13 Sentinel-2 bands or only
+                          the required 10. Default value:  `True`.
+        :type all_bands: bool
+        :param processing_resolution: Resolution to be used during the computation of cloud probabilities and masks.
+                                      Default is `None` (source resolution).
+        :type processing_resolution: str or int
+        :param max_proc_frames: Maximum number of frames (including the target, for multi-temporal classification)
```

Comment:
The number of features computed independent of the number of frames used. The data from all frames is aggregated into statistics. While it makes sense to use the same number of processing frames as the training, a different number of frames can be used without major differences in results.

Reply:
It can be expected that the quality of the statistics will be improved, when more frames are considered at once, otherwise they can be randomly biased in one direction or another (eg. if we're unlucky and have predominantly cloudy images in our batch). Of course, this comes at a cost in terms of processing time and memory consumption, hence the option.

Reply:
got it